packages/data-service/src/connect-mongo-client.ts

```diff
@@ -249,7 +249,12 @@ export async function connectMongoClientDataService({
       connectLogger,
       CompassMongoClient
     );
-    await runCommand(client.db('admin'), { ping: 1 });
+    try {
+      await runCommand(client.db('admin'), { ping: 1 });
+    } catch (err) {
+      await client.close().catch(() => {});
+      throw err;
+    }
```

Comment:
@nbbeeken I think this bit here was always incorrect, so it's good that the driver bump forces us to fix it, but it's still interesting that failing the ping didn't keep the event loop alive in older versions of the driver